umbra-js/src/classes/Umbra.ts

```diff
@@ -231,8 +231,23 @@ export class Umbra {
       // Withdraw ETH
       // Based on gas price, compute how much ETH to transfer to avoid dust
       const ethBalance = await this.provider.getBalance(stealthWallet.address); // stealthWallet.address is our stealthAddress
+
+      // Estimate gas limit if not provided
```

Comment:
One thing we also want to add is a check, probably during setup of `AccountReceiveTableWithdrawConfirmation.vue`, that runs the same `estimateGas` call and updates the fee amount show to the user. Currently if I withdraw to e.g. [0x6851D6fDFAfD08c0295C392436245E5bc78B0185](https://rinkeby.etherscan.io/address/0x6851d6fdfafd08c0295c392436245e5bc78b0185#code), which is a random gnosis safe I found on rinkeby, then the send method here correctly uses the right gas limit, but the estimated fee shown to the user is incorrect.

My suggestion would be to keep the current logic in this file as a fallback, but our frontend should also call `estimateGas`, show the appropriate fee, and pass in the gas limit override to this `withdraw` method

<img src="https://user-images.githubusercontent.com/17163988/151577109-50c2ce7f-a456-456c-ae65-8f91cab4b128.png" width=400 />